btf/types_test.go

```diff
@@ -0,0 +1,73 @@
+package btf
+
+import "testing"
+
+import "fmt"
+
+func TestSizeof(t *testing.T) {
+	testcases := []struct {
+		size int
+		typ  Type
+	}{
+		{1, &Int{Size: 1}},
+		{4, &Enum{}},
+		{0, &Array{Type: &Pointer{Target: Void{}}, Nelems: 0}},
+		{12, &Array{Type: &Enum{}, Nelems: 3}},
+	}
+
+	for _, tc := range testcases {
+		name := fmt.Sprint(tc.typ)
+		t.Run(name, func(t *testing.T) {
+			have := Sizeof(tc.typ)
+			if have != tc.size {
+				t.Errorf("Expected size %d, got %d", tc.size, have)
+			}
+		})
+	}
+}
+
+func TestCopyType(t *testing.T) {
+	_ = copyType(Void{})
+
+	in := &Int{Size: 4}
+	out := copyType(in)
+
+	in.Size = 8
+	if size := out.(*Int).Size; size != 4 {
+		t.Error("Copy doesn't make a copy, expected size 4, got", size)
+	}
+
+	t.Run("cyclical", func(t *testing.T) {
+		ptr := &Pointer{}
+		foo := &Struct{
+			Members: []Member{
+				{Type: ptr},
+			},
+		}
+		ptr.Target = foo
+
+		_ = copyType(foo)
+	})
+}
+
+// The following are valid Types.
+func ExampleType_validTypes() {
```

Comment:
I want to document somehow which (Go) types implement the `Type` interface, since that's not visible from the generated documentation. Any ideas how I can make that more explicit?

Reply:
Maybe just a comment on the interface? Or leave a more elaborate comment on this example explaining what it is for.